Futura/Futura/Either.swift

```diff
@@ -0,0 +1,24 @@
+/* Copyright 2018 Miquido
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License. */
+
+internal enum Result<T> {
+    case success(T)
+    case error(Error)
+}
+
+// TODO: remove Result in favor of Either
+internal enum Either<T, U> {
```

Comment:
This is great, but sometimes I find it hard to read in code when I see `.right` because I have to think about it's type

Reply:
as mentioned above - I will see if keeping Result along with dedicated enum would not be better